packages/solana/src/client.ts

```diff
@@ -441,6 +513,9 @@ export class Web3Modal extends Web3ModalScaffold {
     provider.on('disconnect', disconnectHandler)
     provider.on('accountsChanged', accountsChangedHandler)
     provider.on('connect', accountsChangedHandler)
+    provider.on('auth_rpcRequest', rpcRequestHandler)
+    provider.on('auth_rpcSuccess', rpcSuccessHandler)
+    provider.on('auth_rpcError', rpcErrorHandler)
```

Comment:
where are you emmiting these? why did you end up doing this here? 
do we need to update other clients?

Reply:
in this structure all events are emitted by the providers and the client behavior is handled through these events:

- https://github.com/WalletConnect/web3modal/pull/2709/files/d6f1b9ac0a71d95a6eab05f7181fa597bdccadf2#diff-7292affc7810dedb06b2c96d9e1b0552120ec1f525697b37f93cccb8fafdb2c9R177-R199
- https://github.com/WalletConnect/web3modal/pull/2709/files/d6f1b9ac0a71d95a6eab05f7181fa597bdccadf2#diff-7292affc7810dedb06b2c96d9e1b0552120ec1f525697b37f93cccb8fafdb2c9R153
- https://github.com/WalletConnect/web3modal/pull/2709/files/d6f1b9ac0a71d95a6eab05f7181fa597bdccadf2#diff-7292affc7810dedb06b2c96d9e1b0552120ec1f525697b37f93cccb8fafdb2c9R71
- https://github.com/WalletConnect/web3modal/pull/2709/files/d6f1b9ac0a71d95a6eab05f7181fa597bdccadf2#diff-7292affc7810dedb06b2c96d9e1b0552120ec1f525697b37f93cccb8fafdb2c9R79

The idea is splitting client behavior from provider execution, so we have less replicated code for injected wallet, wallet connect, auth and so on.